src/de/grobox/blitzmail/NoteActivity.java

```diff
@@ -86,6 +90,18 @@ public void onCancel(DialogInterface dialog) {
 		dialog.setCanceledOnTouchOutside(false);
 		dialog.show();
 
+		 ClipboardManager cm = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+		 ClipData cd = cm.getPrimaryClip();
+		 if (cd != null) {
+			 CharSequence text = cd.getItemAt(0).getText();
+			 if (text != null && text.length() > 0) {
+				textView.setText(text.toString());
+				// ensure selectAllOnFocus is triggered
+				textView.clearFocus();
+				textView.requestFocus();
```

Comment:
I think you can call something like `setSelection()` as well to select all text instead of this focus hack. Also this probably only works when the dialog is created once, not when you navigate to another app, copy text there and then come back to the existing activity.

Reply:
Done.